Makefile

```diff
@@ -52,20 +52,18 @@ deps: ## Ensures fresh go.mod and go.sum.
 
 .PHONY: generate
 generate: ## Generate all dynamic files.
-generate: generate-pow generate-sol
+generate: pkg/pow/kernelSource.go pkg/contracts/%/%
 
-.PHONY: generate-pow
-generate-pow:
-	@go run ./scripts/opencl
+pkg/pow/kernelSource.go: scripts/opencl/sources/*
+	go run ./scripts/opencl
 
-.PHONY: generate-sol
-generate-sol: $(SOLCCHECK) $(ABIGEN)
-	@rm -Rf pkg/contracts/
+pkg/contracts/tellor/tellor.go pkg/contracts/getter/getter.go: contracts/* $(SOLCCHECK) $(ABIGEN)
```

Comment:
this means: run this target only if `pkg/contracts/tellor/tellor.go` or  `pkg/contracts/getter/getter.go` are missing or if any of the files in `contracts/*` has changed.

lets ignore the explanation for `$(SOLCCHECK) $(ABIGEN)` for now :+1: